File: BookNestBackend/BookNest.Services/Database/Entities/OrderStatus.cs
using System.ComponentModel.DataAnnotations;

namespace BookNest.Services.Database.Entities
{
    public class OrderStatus
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string Name { get; set; } = string.Empty;

        public ICollection<Order> Orders { get; set; } = new List<Order>();
    }
}

File: BookNestFrontend/booknest_desktop/lib/models/order_status.dart
class OrderStatus {
  final int id;
  final String name;

  OrderStatus({required this.id, required this.name});

  factory OrderStatus.fromJson(Map<String, dynamic> json) {
    return OrderStatus(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
    );
  }
}
